cirq-google/cirq_google/optimizers/two_qubit_gates/gate_compilation.py

```diff
@@ -50,6 +44,7 @@ class TwoQubitGateCompilation(NamedTuple):
 
 
 @dataclass
+@deprecated_class(deadline='v0.16', fix='Stop using.', name='GateTabulation')
 class GateTabulation:
```

Comment:
We can cleanup the deprecation significantly by inheriting the deprecated classes from new classes. For example, you can replace the implementation of `GateTabulation ` by simply:

```python
@deprecated_class(deadline='v0.16', fix='Use cirq.TwoQubitGateTabulation instead')
class GateTabulation(cirq.TwoQubitGateTabulation):
    pass
```

Please do this for all the deprecated classes and cleanup this file.

Reply:
I think this might in theory break some dependent code. Might be safer as a follow-up.